### PR TITLE
Remove moment as a dependency

### PR DESCRIPTION
**Pull request by davidrapson**
We currently use moment to format a single date in our changelog script. We can reasonably use `Intl.DateTimeFormat` here instead. Updates the script to:

- Use `Intl.DateTimeFormat` in place of moment
- Simplify the formatting of heading so there's less special case formatting and so that heading levels aren't skipped awkwardly.

### CODE REVIEW

File: scripts/release.js [hunk 1]
```diff
@@ -127,9 +132,7 @@ prompt([
         const changelog = fs.readFileSync(changelogPath, 'utf8');
 
         // Add the new entry
-        const changelogEntry = `## <sub>v${newVersion}</sub>\n\n#### ${moment().format(
-          '_MMM. D, YYYY_'
-        )}`;
+        const changelogEntry = `## v${newVersion}\n\n### ${formattedDate()}`;
```
Comment (davidsauntson) [inline scripts/release.js:135]: I'm a bit lost, contextually, so I might be being stupid.  This is markdown tho, right? How come there are 2 `#` before and 3 `#` after?  Not that it is really that important!

**Review by davidsauntson — APPROVED** [review]
<whitney houston gif>

File: scripts/release.js (hunk 1 above)
Comment (davidsauntson) [reply scripts/release.js:135]: OMG pls ignore 🙈

Comment (davidrapson) [reply scripts/release.js:135]: Not stupid at all, I was _also_ lost reading this code. Guessing you worked out that this is a h2 and h3. It was super confusing before going 2 to 4.

**Pull request merged**